lib/matplotlib/__init__.py

```diff
@@ -231,7 +231,7 @@ def _check_versions():
 
 # The decorator ensures this always returns the same handler (and it is only
 # attached once).
-@functools.lru_cache()
```

Comment:
Some of these should be converted to functions.cache for clarity, in fact.

Reply:
I updated some of them where I thought an unbounded cache was alright, but others I figured we can handle later if we want to change to unbounded rather than the default size of 128. When I was looking through the `lru_cache` locations it looked like we may even have some decorated instance methods with self that may be keeping instances alive, I wonder if that is causing any of the slowly growing memory leaks that have been reported...

Reply:
I was specifically referring to cases where the cache just serves to establish a singleton, like here.